great_expectations/checkpoint/checkpoint.py

```diff
@@ -99,6 +106,85 @@ def get_substituted_config(
+            validation_batch_request_batch_data: Any = (
+                validation_batch_request.batch_data
+            )
+            validation_batch_request: dict = validation_batch_request.get_json_dict()
+            validation_batch_request["batch_data"] = validation_batch_request_batch_data
+        elif isinstance(validation_batch_request, None):
+            return BatchRequest(**base_batch_request_dict)
+
+        runtime_batch_request_dict = deepcopy(validation_batch_request)
+        for key, val in runtime_batch_request_dict.items():
+            if val is not None and base_batch_request_dict.get(key) is not None:
+                raise CheckpointError(
+                    f"BatchRequest attribute '{key}' was specified in both validation and top-level CheckpointConfig."
+                )
+            runtime_batch_request_dict[key] = base_batch_request_dict[key]
+        return BatchRequest(**runtime_batch_request_dict)
+
+    def validate_validation_dict(self, validation_dict):
+        if validation_dict.get("batch_request") is None:
+            raise CheckpointError("validation batch_request cannot be None")
+        if not validation_dict.get("expectation_suite_name"):
+            raise CheckpointError("validation expectation_suite_name must be specified")
+        if not validation_dict.get("action_list"):
+            raise CheckpointError("validation action_list cannot be empty")
+
+    def get_substituted_validation_dict(
+        self, substituted_runtime_config: CheckpointConfig, validation_dict: dict
+    ) -> dict:
+        substituted_validation_dict = {
+            "batch_request": self.get_runtime_batch_request(
+                substituted_runtime_config=substituted_runtime_config,
+                validation_batch_request=validation_dict.get("batch_request"),
+            ),
+            "expectation_suite_name": validation_dict.get("expectation_suite_name")
+            or substituted_runtime_config.expectation_suite_name,
+            "action_list": CheckpointConfig.get_updated_action_list(
+                base_action_list=substituted_runtime_config.action_list,
+                other_action_list=validation_dict.get("action_list", {}),
+            ),
+            "evaluation_parameters": nested_update(
+                substituted_runtime_config.evaluation_parameters,
+                validation_dict.get("evaluation_parameters", {}),
+            ),
+            "runtime_configuration": nested_update(
+                substituted_runtime_config.runtime_configuration,
+                validation_dict.get("runtime_configuration", {}),
+            ),
+        }
+        if validation_dict.get("name") is not None:
+            substituted_validation_dict["name"] = validation_dict["name"]
+        self.validate_validation_dict(substituted_validation_dict)
+        return substituted_validation_dict
+
+    def get_run_name_from_template(self, run_name_template):
```

Comment:
Might wish to add typehints throughout this method for extra clarity too like we have been everywhere else...